dist2src.py

```diff
@@ -212,7 +214,7 @@ def extract_archive(ctx, origin, dest):
     with tempfile.TemporaryDirectory() as tmpdir:
         shutil.unpack_archive(archive, tmpdir)
         # Expect an archive with a single directory.
-        assert len(os.listdir(tmpdir)) == 1
+        assert len(os.listdir(tmpdir)) == 1, "Archive content is not a single directory"
```

Comment:
I know it's a script but still, `asserts` should not be used outside of the tests. (When running python code in optimise mode using e.g. `-O`, it's ignored.)

(I know you only added a message to the assert but would be nice to avoid that usage.)